omnipaxos_storage/src/memory.rs

```diff
@@ -0,0 +1,374 @@
+            c_opts.index_max_items(5000); // Max 10,000 log entries in the commitlog
+            c_opts.message_max_bytes(64000); // Max 64 kilobytes for each message
+
+            let mut db_opts = Options::default();
+            db_opts.set_write_buffer_size(0); // Set amount data to build up in single memtable before converting to on-disk file
+            db_opts.set_db_write_buffer_size(0); // Set total amount of data to build up in all memtables before writing to disk
+            db_opts.set_compression_type(rocksdb::DBCompressionType::Lz4); // set compression type
+            db_opts.set_max_write_buffer_number(1); // Max buffers for writing in memory
+            db_opts.set_max_write_buffer_size_to_maintain(0); // max size of write buffers to maintain in memory
+            db_opts.set_use_direct_reads(true); // data read from disk will not be cache or buffered
+            db_opts.set_use_direct_io_for_flush_and_compaction(true); // data flushed or compacted will not be cached or buffered
+            db_opts.increase_parallelism(4); // Set the amount threads for rocksDB compaction and flushing
+            db_opts.create_if_missing(true); // Creates an database if its missing in the path
+
+            (c_opts, db_opts)
+        }
+    }
+
+    impl<T, S> Storage<T, S> for PersistentState<T, S>
+    where
+        T: Entry + AsBytes + FromBytes,
+        S: Snapshot<T>,
+    {
+        fn append_entry(&mut self, entry: T) -> u64 {
+            //println!("append entry {:?}", entry);
+            let entry_bytes = AsBytes::as_bytes(&entry);
+            match self.c_log.append_msg(entry_bytes) {
+                Ok(x) => x,
+                Err(_e) => 0,
+            }
+        }
+
+        fn append_entries(&mut self, entries: Vec<T>) -> u64 {
+            //println!("append entries!");
+            let mut buf: MessageBuf = MessageBuf::default();
+            for entry in entries {
+                let _ = buf.push(AsBytes::as_bytes(&entry));
+            }
+            self.c_log.append(&mut buf).unwrap();
+            self.get_log_len()
+        }
+
+        fn append_on_prefix(&mut self, from_idx: u64, entries: Vec<T>) -> u64 {
+            //println!("append on prefix!");
+            let _ = self.c_log.truncate(from_idx);
+            self.append_entries(entries)
+        }
+
+        fn get_entries(&self, from: u64, to: u64) -> Vec<T> {
+            println!("get_entries from: {:?} -> to: {:?} ", from, to);
+            let buffer = self
+                .c_log
+                .read(from, ReadLimit::default())
+                .unwrap_or(MessageBuf::default());
+            let mut entries = vec![];
+            let zero_idx_to = to - from;
+            for (idx, msg) in buffer.iter().enumerate() {
+                if (idx as u64) >= zero_idx_to {
+                    break;
+                }
+                entries.push(FromBytes::read_from(msg.payload()).unwrap());
+            }
```

Comment:
This loop looks a bit confusing/over-complicated. You can probably do something like this:
```
let iter = buffer.iter();
for _ in from..to {
    let msg = iter.next();
    entries.push(FromBytes::read_from(msg.payload()).unwrap());
} 
```